tensorboard/summary/writer/event_file_writer_test.py

```diff
@@ -49,127 +49,138 @@ def test_event_file_writer_roundtrip(self):
     event_files = sorted(glob.glob(os.path.join(logdir, '*')))
     self.assertEqual(len(event_files), 1)
     r = PyRecordReader_New(event_files[0])
-    r.read()
-    events = r.event_strs
-    event_from_disk = events[1]
-    assert fakeevent.SerializeToString() == event_from_disk
+    r.GetNext()  # meta data, so skip
+    r.GetNext()
+    self.assertEqual(fakeevent.SerializeToString(), r.record())
 
   def test_setting_filename_suffix_works(self):
-    logdir = tempfile.mkdtemp()
+    logdir = self.get_temp_dir()
 
     w = EventFileWriter(logdir, filename_suffix='.event_horizon')
     w.close()
     event_files = sorted(glob.glob(os.path.join(logdir, '*')))
-    assert event_files[0].split('.')[-1] == 'event_horizon'
+    self.assertEqual(event_files[0].split('.')[-1], 'event_horizon')
+
+
+class AsyncWriterTest(tb_test.TestCase):
+  def __init__(self, *args, **kwargs):
+    super(AsyncWriterTest, self).__init__(*args, **kwargs)
 
   def test_async_writer_without_write(self):
     for i in range(100):
-      logdir = tempfile.mkdtemp()
+      logdir = self.get_temp_dir()
       w = EventFileWriter(logdir)
```

Comment:
This test seems misplaced - it's using `EventFileWriter` so it should go in the other TestCase above.  Also I'm not sure wrapping it in the `for i in range(100)` adds anything - can we just remove the repetition?

It might also make sense to just add the lines

```python
s = event_pb2.Event.FromString(r.record())
self.assertEqual(s.file_version, "brain.Event:2")
```
to the existing `test_event_file_writer_roundtrip` above, after the first `r.GetNext()`.